### PR TITLE
add support for kubernetes authentication backend following => https:…

### PR DESCRIPTION
**Pull request by jeffwecan**
This PR is meant to replace https://github.com/ianunruh/hvac/pull/158 and builds on the proposed changes in that PR's branch.

Original PR's description:

> This implement in HVAC library the kubernetes auth backend API described here => https://www.vaultproject.io/api/auth/kubernetes/index.html .

### CODE REVIEW

File: hvac/tests/test_kubernetes_methods.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    def test_auth_kubernetes(self, test_label, test_role, test_jwt, mount_point, requests_mocker):
+        mock_response = {
+            'auth': {
+                'accessor': 'accessor-1234-5678-9012-345678901234',
+                'client_token': 'cltoken-1234-5678-9012-345678901234',
+                'lease_duration': 10000,
+                'metadata': {
+                    'role': 'custom_role',
+                    'service_account_email': 'dev1@project-123456.iam.gserviceaccount.com',
+                    'service_account_id': '111111111111111111111'
```
Comment (seanmalloy) [inline hvac/tests/test_kubernetes_methods.py]: This mocked response does not look correct to me. The `service_account_email` and `service_account_id` fields are from the GCP auth backed and not the Kubernetes auth backend.

I believe the correct field names can be found in the Vault API docs at the below URL.

https://www.vaultproject.io/api/auth/kubernetes/index.html#sample-response-3

**Comment by seanmalloy** [comment]
I think it would be a good idea to add an example to the README of how to use the `auth` method from within a Kubernetes pod.

**Comment by seanmalloy** [comment]
@jeffwecan i stubbed in the k8s auth docs into the README and pushed up the commit to your branch. I'll try to get my kube cluster working with this code and get the proper docs created.

File: hvac/tests/test_kubernetes_methods.py (hunk 1 above)
Comment (seanmalloy) [reply hvac/tests/test_kubernetes_methods.py]: I just pushed up the fix for this to your branch.

**Comment by seanmalloy** [comment]
@jeffwecan I was able to successfully test the `auth_kubernetes` method with a real Kubernetes clusters, and I pushed up the docs.

**Comment by jeffwecan** [comment]
Sweet! Thanks for filling in those last few bits. Since it sounds like you've validated these changes with a real world cluster, I'll go ahead and merge this in. I'm going to see about merging in one other PR this morning and then I'll cut a new release with these changes plus your previously merged in GCP method.

**Pull request merged**